lib/3scale/backend/user.rb

```diff
@@ -142,6 +159,17 @@ def self.clear_cache(service_id, user_id)
         Memoizer.clear key
       end
 
+      def self.delete_users_in_batch(service_id, usernames)
+        if usernames.size != 0
+          service_usernames_keys = usernames.map { |username| self.key(service_id, username) }
+          storage.del(service_usernames_keys)
```

Comment:
I think these 2 should be grouped in a pipeline:
```ruby
          storage.pipelined do
            storage.del(service_usernames_keys)
            storage.srem(self.service_users_set_key(service_id), usernames)
          end
```

Sorry, I didn't see this in my first review. All the rest looks good to me :+1:

Reply:
Changed